### ISSUE TITLE
Compilation warning issue for AQMv7 implementation.

### ISSUE THREAD

**Issue by JianpingHuang-NOAA** [NOAA-EMC/UPP#817]
@WenMeng-NOAA

The AQMv7 package was delivered on Oct. 31 but was retuned due to the violation of the EE2 compliance. UPP-related warnings are part of them. 

We use hash = 22cfb88 to support the AQMv7 implementation

The UPP compilation warning message can be found on WCOSS2 from the following file
/lfs/h2/emc/physics/noscrub/jianping.huang/nwdev/packages/aqm.v7.0.88c/sorc/warnings.log 

, and the UPP-related warnings were copied below

/lfs/h2/emc/physics/noscrub/jianping.huang/nwdev/packages/aqm.v7.0.88c/sorc/UPP/sorc/ncep_post.fd/INITPOST_GFS_NEMS_MPIIO.f(30): warning #16219: Some OpenMP processing was skipped to constrain compile time. Consider overriding limits (-qoverride-limits)./lfs/h2/emc/physics/noscrub/jianping.huang/nwdev/packages/aqm.v7.0.88c/sorc/UPP/sorc/ncep_post.fd/INITPOST_NETCDF.f(35): warning #16219: Some OpenMP processing was skipped
 to constrain compile time. Consider overriding limits (-qoverride-limits)./lfs/h2/emc/physics/noscrub/jianping.huang/nwdev/packages/aqm.v7.0.88c/sorc/UPP/sorc/ncep_post.fd/io_int_stubs.f(4): warning #6843: A dummy argument with an explicit INTEN
T(OUT) declaration is not given an explicit value.   [STATUS]/lfs/h2/emc/physics/noscrub/jianping.huang/nwdev/packages/aqm.v7.0.88c/sorc/UPP/sorc/ncep_post.fd/io_int_stubs.f(4): warning #6843: A dummy argument with an explicit INTEN
T(OUT) declaration is not given an explicit value.   [DATA]

Please help us to fix the warnings as soon as possible.  Let me know if you need more information on AQMv7 package.  Thanks  !

@

**Comment by WenMeng-NOAA** [comment]
@JianpingHuang-NOAA A release branch "release/aqm_v7" was created at the UPP repository. You may use this branch for the UPP code checkout in AQM V7 package. The UPP has a build script under UPP/tests/compile_upp.sh to generate an executable upp.x under UPP/exec/. 
You may look at my building log compile.log at /u/wen.meng/noscrub/aqm_imp/UPP/tests on dogwood. There are no warning message for io_int_stubs.f.

**Comment by WenMeng-NOAA** [comment]
@JianpingHuang-NOAA To address your issue for AQM V7 implementation:

- Regarding to OpenMP warnings 
  A fix proposed by @DusanJovic-NOAA was incorporated in the branch release/aqm_v7 at @54cbb57 to solve them.
- Regarding to warning 
   **[100%] Linking Fortran executable upp.x
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: warning: libjpeg.so.62, needed by /apps/spack/jasper/2.0.25/intel/19.1.3.304/sjib74krrorkyczqpqah4tvewmlnqdx4/lib64/libjasper.so, may conflict with libjpeg.so.9**
Dusan's comments:

>  I do not think that there's any change in upp's CMakeLists.txt file(s) that we can/should make to eliminate this warning. This warning complains about potential shared library mismatch. So I think this is either a spack library or system library issue. I would ask NCO about this warning and ask them what they suggest. Ideally the jasper library in /apps/spack/jasper should be recompiled to use the correct libjpeg.so, in my opinion.

- Regarding to warning **[STATUS]/lfs/h2/emc/physics/noscrub/jianping.huang/nwdev/packages/aqm.v7.0.88c/sorc/UPP/sorc/ncep_post.fd/io_int_stubs.f(4): warning #6843: A dummy argument with an explicit INTEN
T(OUT) declaration is not given an explicit value. [DATA]**
You may work with SRW developer to add some CMAKE changes in the UPP build procedure in SRW. You can provide him with the UPP build script UPP/tests/compile_upp.sh

Loop @HuiyaChuang-NOAA and @yangfanglin for their awareness.

**Comment by JianpingHuang-NOAA** [comment]
@RaghuReddy-NOAA @rmontuoro  Raghu, this is the issue that we mentioned for UPP and other codes like UFS_UTILS

Linking Fortran executable upp.x
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: warning: **libjpeg.so.62**, needed by /apps/spack/jasper/2.0.25/intel/19.1.3.304/sjib74krrorkyczqpqah4tvewmlnqdx4/lib64/libjasper.so, may conflict with **libjpeg.so.9**

Do you have any idea(s) on that?


Jianping

**Comment by JianpingHuang-NOAA** [comment]
@chan-hoo  Can you help us to modify the devbuild.sh file for using the same settings as the ones used by  compile_upp.sh  at  /lfs/h2/emc/physics/noscrub/jianping.huang/nwdev/packages/aqm.v7.0.88c/sorc/UPP/tests (Cactus).

Meanwhile, please update the hash number with **54cbb57** for UPP in support of AQMv7 implementation.

**Comment by chan-hoo** [comment]
@WenMeng-NOAA, can you copy your `compile.log` file to Cactus? If it is not available now, can you let me know which build options should be added to the cmake file of AQM.v7 for UPP? `-DBUILD_WITH_GTG`, `-DBUILD_WITH_WRFIO`, `-DINTERNAL_IFI`, or/and `-DREQUIRE_IFI`?

**Comment by JianpingHuang-NOAA** [comment]
@chan-hoo  You can find compile_upp.sh and compile.log file for UPP on Cactus

/lfs/h2/emc/physics/noscrub/jianping.huang/nwdev/packages/aqm.v7.0.88c/sorc/UPP/tests

**Comment by chan-hoo** [comment]
@JianpingHuang-NOAA, I've updated `CMakeLists.txt` and hash of UPP in `production/AQM.v7`: https://github.com/ufs-community/ufs-srweather-app/commit/fb7c9d6c9851146be3488b68da09774602b3ac79
I can't see any warnings from UPP except for libjpeg.so:
```
/lfs/h2/emc/lam/noscrub/chan-hoo.jeon/prod_aqm_test/ufs-srweather-app/sorc/build/log.make
```

**Comment by JianpingHuang-NOAA** [comment]
@chan-hoo @WenMeng-NOAA  Great ! I will test it. Thanks.

**Comment by WenMeng-NOAA** [comment]
This issue was solved in PR #818.
